valheim-backup

```diff
@@ -23,6 +23,8 @@ main() {
         while [ $run = true ]; do
             backup
             flush_old
+	    sleep 2
```

Comment:
What is this 2 second sleep for?

Reply:
waiting on harddrive syncing

Reply:
This is not a thing. So first, if you wanted to make the kernel flush its buffers to disk you could call `sync`. That is more reliable than sleeping for two seconds. There is no guarantee that fs buffers have been flushed after 2s.
But secondly this really is not needed at all since you are using user space tools like rsync which use libc functions like `open()` and `read()` which in turn use the corresponding kernel functions (syscalls) like `open()` and `read()`. So even if the zip file is in full or partially in a fs buffer that hasn't been flushed to disk yet there is no way for you to not receive the correct data when accessing the file using standard user space tools.
Only if you were to access the raw block device directly, bypassing the kernel's interfaces for accessing the filesystem, would it make any sense to sync buffers to disk. I.e. never.